include/core/dimensionality.h

```diff
@@ -0,0 +1,107 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2019 -  by the Lethe authors
+ *
+ * This file is part of the Lethe library
+ *
+ * The Lethe library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+#ifndef lethe_dimensionality_h
+#define lethe_dimensionality_h
+
+#include <deal.II/base/parameter_handler.h>
+
+using namespace dealii;
+
+/**
+ * @brief Dimensionality - This is used to rescale physical properties for cases where the main dimensions
+ * of the problems (Length, Mass, Time) are not expressed in SI, but are
+ * expressed in another set of unit. For example, this can be used to carry
+ * out a simulation where the mesh is expressed in millimeter instead of
+ * meters. This class is currently only used to calculate to rescale the
+ * physical properties in the FEM-based physics (e.g. all of those that are
+ * solvers). Support for DEM, CFD-DEM and rescaling of boundary conditions is
+ * planned in the mid-term.
+ */
+namespace Parameters
+{
+  class Dimensionality
+  {
+  public:
+    /**
+     * @brief
+     * Default constructor that assumes that the units used in the simulation
+     * are consistent.
+     */
+    Dimensionality()
+      : length(1)
+      , mass(1)
+      , time(1)
+      , temperature(1)
+    {
+      define_all_scales();
+    }
+
+    /**
+     * @brief
+     * Defines all the sub scales that are relevant to simulations. This is used
+     * to define scaling laws for kinematic viscosity or other quantities which
+     * appear multiple times in the simulation. This prevents the recalculation
+     * of these scales (and thus, human errors).
+     */
+    void
+    define_all_scales();
+
+    /**
+     * @brief
+     * Declares the fundamental dimensions in the parameter file
+     *
+     * @param prm The parameter handler being used
+     */
+    void
+    declare_parameters(ParameterHandler &prm);
+
+    /**
+     * @brief
+     * Parses the fundamental dimensions from the parameter file
+     *
+     * @param prm The parameter handler being used
+     */
+    void
+    parse_parameters(ParameterHandler &prm);
+
+    /*
+     * Fundamental dimensions. These are the dimensions that are read from the
+     * user
+     */
+    double length;
+    double mass;
+    double time;
+    double temperature;
```

Comment:
I think it would be useful to have similar names between the fundamental dimensions and the scaling laws. I would rename the member "length" to "length_scaling", for example, or remove the "_scaling" from the scaling law members.

Reply:
I was not really sure about this because they have both very different meaning.
When you say Length = 0.001 it means that your fundamental length is 1mm
But when you calculate the viscosity scaling, it's actually the opposite. You are calculating the scaling factor to transform a viscosity in m^2/s to mm^2 /s. So in essence, it's 1 over the other if you know what i mean. It's not L*L/T, but actually T/L/L. Hence why I wanted to use the scaling suffix, to differentiate between the fundamental quantities AND the scaling factors that arise from them.

Reply:
I didn't see it like that at first, but it makes sense. I don't see a better/simpler way to do this for now